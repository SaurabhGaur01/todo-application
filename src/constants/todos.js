export const ADD_TODO = 'modules/todo-app/ADD_TODO';
export const CLEAR_TODO = 'modules/todo-app/CLEAR_TODO';
export const MASS_ADD = 'modules/todo-app/MASS_ADD';
export const TOGGLE_TODO = 'modules/todo-app/TOGGLE_TODO';