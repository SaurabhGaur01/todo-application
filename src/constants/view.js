export const VIEW_SHOW_ALL = 'SHOW_ALL';
export const VIEW_SHOW_COMPLETED = 'SHOW_COMPLETED';
export const VIEW_SHOW_ACTIVE = 'SHOW_ACTIVE';