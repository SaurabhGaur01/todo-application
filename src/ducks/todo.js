import { ADD_TODO, CLEAR_TODO, MASS_ADD, TOGGLE_TODO } from '../constants/todos';

let nextTodoId = 0
export const addTodo = (text) => ({
  type: ADD_TODO,
  id: nextTodoId++,
  text
});

export const clearTodo = () => ({
  type: CLEAR_TODO
});

export const massAdd = data => ({
  type: MASS_ADD,
  data
});

export const toggleTodo = (id) => ({
    type: TOGGLE_TODO,
    id
});