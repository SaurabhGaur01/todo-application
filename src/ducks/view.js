import { VIEW_SHOW_ALL } from '../constants/view';

export const SET_VIEW = 'modules/todo-app/SET_VIEW';

export const setView = (filter) => ({
    type: SET_VIEW,
    filter
})

export const viewReducer = (state = VIEW_SHOW_ALL, action = { type: 'NULL_ACTION'}) => {
    switch (action.type) {
      case SET_VIEW:
        return action.filter
      default:
        return state
    }
}