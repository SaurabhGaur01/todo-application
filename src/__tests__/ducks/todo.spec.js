import {
    addTodo,
    clearTodo,
    toggleTodo,
    massAdd
} from '../../ducks/todo';

import { todosReducer } from '../../ducks/todos';
import { MASS_ADD, ADD_TODO } from '../../constants/todos';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

const mockKey = 'mockValue';

describe('todo Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('addTodo should match the snapshot' , () => {
            expect(addTodo(mockKey)).toMatchSnapshot();
        });
        it('clearTodo should match the snapshot' , () => {
            expect(clearTodo()).toMatchSnapshot();
        });
        it('toggleTodo should match the snapshot' , () => {
            expect(toggleTodo(mockKey)).toMatchSnapshot();
        });
        it('massAdd should match the snapshot' , () => {
            expect(massAdd(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(todosReducer('', { type: ''})).toEqual('');
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(todosReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should clear the store when called with clearRepoData' , () => {
            expect(todosReducer(mockObject, clearTodo())).toEqual([]);
        });
    });
});