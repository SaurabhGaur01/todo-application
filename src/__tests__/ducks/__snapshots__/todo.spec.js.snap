// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo Duck Tests Action Generator addTodo should match the snapshot 1`] = `
Object {
  "id": 0,
  "text": "mockValue",
  "type": "modules/todo-app/ADD_TODO",
}
`;

exports[`todo Duck Tests Action Generator clearTodo should match the snapshot 1`] = `
Object {
  "type": "modules/todo-app/CLEAR_TODO",
}
`;

exports[`todo Duck Tests Action Generator massAdd should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/todo-app/MASS_ADD",
}
`;

exports[`todo Duck Tests Action Generator toggleTodo should match the snapshot 1`] = `
Object {
  "id": "mockValue",
  "type": "modules/todo-app/TOGGLE_TODO",
}
`;
